backend/httpresource/handler.go

```diff
@@ -71,5 +72,22 @@ func PluginConfigFromContext(ctx context.Context) backend.PluginConfig {
 	if v == nil {
 		return backend.PluginConfig{}
 	}
+
 	return v.(backend.PluginConfig)
 }
+
+type userKey struct{}
+
+func withUser(ctx context.Context, cfg *backend.User) context.Context {
+	return context.WithValue(ctx, userKey{}, cfg)
+}
+
+// UserFromContext returns user from context.
+func UserFromContext(ctx context.Context) *backend.User {
+	v := ctx.Value(userKey{})
+	if v == nil {
```

Comment:
Since a pointer to `backend.User` is returned I think nil should be returned if it's nil. Otherwise if there should always be a user present then change to non-pointer in the sdk contracts.

Speaking about alerting/Grafana doing requests to backend plugin without a user, is the idea that this is the only time user will be nil in a backend plugin?

Reply:
:+1: 

I'm not sure about Grafana Alerting yet. The user could either be nil or have a field that indicates that it's alerting. But I think it can be nil for now. What do you think?

Reply:
How about we use nil for alerting for now and we can revisit this later?

Reply:
Sounds good :+1: